git-flow/github-actions.md

```diff
@@ -0,0 +1,721 @@
+# GitHub Actions Documentation
+
+GitHub Actions is a powerful automation and continuous integration/continuous deployment (CI/CD) platform provided by GitHub. It allows you to define custom workflows that automate various tasks in your software development process. This documentation will guide you through the basics of using GitHub Actions in your repository.
+
+## Table of Contents
+
+1. [Introduction](#introduction)
+2. [Getting Started](#getting-started)
+3. [Workflow Structure](#workflow-structure)
+4. [Triggering Workflows](#triggering-workflows)
+5. [Using Marketplace Actions](#using-marketplace-actions)
+6. [Secrets](#secrets)
+7. [Debugging Workflows](#debugging-workflows)
+8. [Example Workflows](#best-practices)
+9. [Conclusion](#conclusion)
+
+## 1. Introduction
+
+GitHub Actions enables you to automate tasks in your GitHub repositories, such as building, testing, and deploying your code. You can define custom workflows using YAML syntax, which are executed based on various events or triggers within your repository.
+
+## 2. Getting Started
+
+To get started with GitHub Actions:
+
+1. **Enable Actions:** Go to the "Actions" tab in your GitHub repository and follow the prompts to enable Actions.
+
+2. **Create a Workflow:** In your repository, create a new directory called `.github/workflows`. Inside this directory, create a new YAML file, e.g., `example.yml`, and define your workflow in it.
+
+3. **Write a Workflow:** Define your workflow with jobs, steps, and actions. Use the available actions and tools to build, test, and deploy your code.
+
+## 3. Workflow Structure
+
+A workflow consists of one or more jobs, and each job consists of one or more steps. Steps are individual tasks that execute in sequence. You can use existing actions or write your own custom actions to define the steps.
+
+A GitHub Actions workflow is defined using YAML syntax and consists of several key components that define the automation tasks to be performed. Below is an overview of the workflow structure:
+
+### 3.1 Workflow Name
+
+The name of the workflow is used to identify it in the GitHub Actions dashboard and can be any descriptive string.
+
+```yaml
+name: My Workflow
+```
+
+### 3.2 Events (on)
+
+The `on` keyword specifies the events that trigger the workflow. These events can be actions like push, pull requests, issue comments, scheduled events, etc.
+
+```yaml
+on:
+  push:
+    branches:
+      - main
+      - feature/*
+  pull_request:
+    types:
+      - opened
+      - synchronize
+```
+
+### 3.3 Jobs
+
+A workflow can consist of one or more jobs. Each job is a set of steps that execute on the same runner. Jobs run in parallel by default but can also be defined to run sequentially.
+
+```yaml
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

Comment:
I'd suggest adding more info about where we can run jobs (not only `ubuntu-latest`)